### PR TITLE
feat: Add javascript and typescript code in chapter_computational_com…

### PR DESCRIPTION
**Pull request by wenjianmin**
…plexity

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

console outputs：
```
$ node recursion.js 
递归函数的求和结果 res = 15
尾递归函数的求和结果 res = 15
斐波那契数列的第 5 项为 3
递归转化为迭代的求和结果为 res = 15
````

### CODE REVIEW

**Review by krahets — COMMENTED** [review]
The code comments are not consistent with other languages. Please address it.

**Comment by wenjianmin** [comment]
> The code comments are not consistent with other languages. Please address it.

OK，thanks for reminding！

File: codes/javascript/chapter_computational_complexity/recursion.js [hunk 1]
```diff
@@ -65,4 +65,4 @@ res = fib(n);
 console.log(`斐波那契数列的第 ${n} 项为 ${res}`);
 
 res = forLoopRecur(n);
-console.log(`递归转化为迭代的求和结果为 res = ${res}`);
+console.log(`递归转化为迭代的求和结果 res = ${res}`);
```
Comment (justin-tse) [inline codes/javascript/chapter_computational_complexity/recursion.js:68]: The code comments are not consistent with other languages. Please address it.

File: codes/typescript/chapter_computational_complexity/recursion.ts [hunk 2]
```diff
@@ -64,6 +64,6 @@ res = fib(n);
 console.log(`斐波那契数列的第 ${n} 项为 ${res}`);
 
 res = forLoopRecur(n);
-console.log(`递归转化为迭代的求和结果为 res = ${res}`);
+console.log(`递归转化为迭代的求和结果 res = ${res}`);
```
Comment (justin-tse) [inline codes/typescript/chapter_computational_complexity/recursion.ts:67]: The code comments are not consistent with other languages. Please address it.

**Review by justin-tse — COMMENTED** [review]
Thanks @wenjianmin Please make sure the function forLoopRecur order are the same with Java or Python

**Review by justin-tse — COMMENTED** [review]
@wenjianmin Thank you for your contribution.

**Review by krahets — APPROVED** [review]
Thanks @wenjianmin for the PR and thx Justin for the review!

**Pull request merged**